### ISSUE TITLE
Undo works only partially

### ISSUE THREAD

**Issue by MURTOMAASORTAJA** [ryanmcalister/unotes#147]
As the title states, Ctrl+Z or Edit -> Undo doesn't undo things that involve other than manually typed stuff, 
i.e. I can undo manually typed stuff, keystroke by keystroke, but a deletion of a chunk of selected text, or pasting stuff from clipboard is un-undoable. I can repro this both in WYSIWYG and in splitview thingie.

If I type "abcd" with my keyboard, paste a paragraph of text from elsewhere and then slap Ctrl+Z a few times, it's the abcd-text that gets undone, letter by letter.

Happy to provide additional information if requested.

Using VS Code 1.73.1, UNotes 1.5.0 and Win11 Enterprise 21H2 22000.1219

**Comment by Orionde** [comment]
I have the exact same issue.

It also looks like that I can't undo anything related to tables in WYSIWYG. If I create a new row then Ctrl + Z, it will not undo the row creation. 

It would be really nice if this issue could be fixed as the undo feature is indispensable. 

Using VS Code 1.77.1, Unotes v1.5.2 and Ubuntu 22.04 x64 5.19.0-38-generic

**Comment by Tsabary** [comment]
Yeah, it's a real pain. I'm not sure why, but a couple of times where I hit backspace to delete an unnecessary new line, it deleted the entire paragraph proceeding it, and I completely had to re-write it which is extremely frustrating. No amount of hitting ctrl+z helps, it's gone.
